modules/nf-commons/src/main/nextflow/plugin/PluginsFacade.groovy

```diff
@@ -95,6 +98,17 @@ class PluginsFacade implements PluginStateListener {
         }
     }
 
+    protected String getPluginsIndexUrl() {
+        final url = env.get('NXF_PLUGINS_URL')
+        if( url ) {
+            log.trace "Detected NXF_PLUGINS_URL=$url"
+            return url
+        } else {
+            log.trace "Using default plugins url"
```

Comment:
We may want show warning when using a non-official index url, in the same vein of 

https://github.com/nextflow-io/nextflow/blob/3e7670f5486c1465bc3547448f27a1b7e0bdeaa0/modules/nf-commons/src/main/nextflow/plugin/PluginUpdater.groovy#L99-L112